.travis.yml

```diff
@@ -0,0 +1,27 @@
+language: php
+
+php:
+  - 7.0
+  - 7.1
+  - 7.2
+  - nightly
+
+matrix:
+  allow_failures:
+    - php: nightly
+
+before_script:
+  - wget https://getcomposer.org/composer.phar
+  - php composer.phar install --no-interaction
+  - composer self-update
+  - git clone https://github.com/octobercms/october.git
+  - mkdir -p ./october/plugins/rainlab/user
+  - mv !(october) october/plugins/rainlab/user
+  - cd october
+  - composer install
+  - cp config/cms.php config/testing/cms.php
+  - cd plugins/rainlab/user
+  - composer install
```

Comment:
@LukeTowers would calling composer update from the root directory be essentially the same as running composer install from both directories? I'm looking for ways to simplify this, and all I'm really seeing is collapsing this to one statement

Reply:
Yes it would, the wikimedia merge plugin takes care of that

Reply:
Ok cool, then I can at least remove a line here. Beyond that though, not sure how much more can be simplified. I'll throw that last commit on this PR in the next day or so. Do you think we'll be able to merge it after that is done, or does it need more work?

Reply:
Possibly, I just need to find the time to review it closely and merge it. I want this to set the standard for the other RainLab plugins (and OctoberCMS plugins in general) so it's important to to me that it's done well. What are your thoughts on including the composer.lock file? I know the main October repo doesn't but what are your thoughts on doing it on a plugin level?

Reply:
Also I'm traveling for the next week (and have been for the past month) so may be slightly delayed :)

Reply:
I agree, this one needs to be as good as it can possibly be. With that in mind, once this is merged and integrated with travis, we can work on an integration with codecov or coveralls to demonstrate that as well, With regard to the lock file, I always think it's a good practice to commit those.

Reply:
Does that leave the opportunity for it to conflict with October's dependencies at some point though? Since it's loaded along with the project and all other plugin's composer files at once?

Reply:
I'm not the best person to speak to that, I've always committed them just because I heard that it's a good practice. I'm not sure how accurate that is in situations where you have multiple lock files.

Reply:
In this case I think that I'd prefer that composer.lock files aren't committed, just to avoid the potential for conflicts with a project level use of composer